File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.cpp
#include <string>
#include <vector>
#include <set>

using namespace std;

vector<int> solution(vector<int> numbers) 
{
    set<int> set;
    
    for(int i=0; i<numbers.size() - 1; i++)
    {
        for(int j=i+1; j<numbers.size(); j++)
        {
            set.insert(numbers[i] + numbers[j]);
        }
    }
    
    vector<int> ans;
    for(auto s : set)
    {
        ans.push_back(s);
    }
    return ans;
}

File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.cs
using System;
using System.Collections.Generic;

public class Solution 
{
    public int[] solution(int[] numbers) 
    {
        HashSet<int> numSet = new HashSet<int>();
        
        for(int i=0; i<numbers.Length - 1; i++)
        {
            for(int j = i+1; j<numbers.Length; j++)
            {
                numSet.Add(numbers[i] + numbers[j]);
            }
        }
        
        List<int> result = new List<int>();
        foreach(int n in numSet)
        {
            result.Add(n);
        }
        result.Sort();
        return result.ToArray();
    }
}